src/msw/listctrl.cpp

```diff
@@ -1494,12 +1499,31 @@ bool wxListCtrl::EnableCheckBoxes(bool enable)
 
 void wxListCtrl::CheckItem(long item, bool state)
 {
-    ListView_SetCheckState(GetHwnd(), (UINT)item, (BOOL)state);
+    if ( !HasCheckBoxes() )
+        return;
+
+    if ( IsVirtual() )
+    {
+        wxListEvent le(state ? wxEVT_LIST_ITEM_CHECKED : wxEVT_LIST_ITEM_UNCHECKED, GetId());
+        le.SetEventObject(this);
+        le.m_itemIndex = item;
+        GetEventHandler()->ProcessEvent(le);
+    }
+    else
+    {
+        ListView_SetCheckState(GetHwnd(), (UINT)item, (BOOL)state);
+    }
 }
 
 bool wxListCtrl::IsItemChecked(long item) const
 {
-    return ListView_GetCheckState(GetHwnd(), (UINT)item) != 0;
+    if ( !HasCheckBoxes() )
+        return false;
```

Comment:
Not sure about this one. Sometimes it is easier to just call `IsItemChecked()` even if checkboxes are disabled. Like I do now in the code below. This would have to be changed to `HasCheckBoxes() && IsItemChecked(line)`.

https://github.com/wxWidgets/wxWidgets/blob/f68bb391ce844ac1805394719758fb7b101aadb7/src/generic/listctrl.cpp#L2095-L2100

Reply:
OK, thanks, let's leave it like this, but I'll add a note about this (not necessarily obvious) behaviour to the docs.

Reply:
Oops, actually this is already documented, sorry for not realizing this.